api/hypershift/v1beta1/azure.go

```diff
@@ -52,33 +46,20 @@ type AzureNodePoolPlatform struct {
 	// +kubebuilder:validation:Required
 	Image AzureVMImage `json:"image"`
 
-	// DiskSizeGB is the size in GB to assign to the OS disk
-	// CAPZ default is 30GB, https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/b3708019a67ff19407b87d63c402af94ca4246f6/api/v1beta1/types.go#L599
-	//
-	// +kubebuilder:default:=30
-	// +kubebuilder:validation:Minimum=16
-	// +optional
-	DiskSizeGB int32 `json:"diskSizeGB,omitempty"`
-
-	// DiskStorageAccountType is the disk storage account type to use. Valid values are:
-	// * Standard_LRS: HDD
-	// * StandardSSD_LRS: Standard SSD
-	// * Premium_LRS: Premium SDD
-	// * UltraSSD_LRS: Ultra SDD
-	//
-	// Defaults to Premium_LRS. For more details, visit the Azure documentation:
-	// https://docs.microsoft.com/en-us/azure/virtual-machines/disks-types#disk-type-comparison
-	//
-	// +kubebuilder:default:=Premium_LRS
-	// +kubebuilder:validation:Enum=Standard_LRS;StandardSSD_LRS;Premium_LRS;UltraSSD_LRS
-	// +optional
-	DiskStorageAccountType string `json:"diskStorageAccountType,omitempty"`
+	// osDisk provides configuration for the OS disk for the nodepool.
+	// This can be used to configure the size, storage account type, encryption options and whether the disk is persistent or ephemeral.
+	// When not provided, the platform will choose reasonable defaults which are subject to change over time.
+	// Review the fields within the osDisk for more details.
+	OSDisk AzureNodePoolOSDisk `json:"osDisk"`
 
-	// AvailabilityZone is the failure domain identifier where the VM should be attached to. This must not be specified
-	// for clusters in a location that does not support AvailabilityZone.
+	// availabilityZone is the failure domain identifier where the VM should be attached to.
+	// This must not be specified for clusters in a location that does not support AvailabilityZone because... TODO: why?
+	// Availability zones are identified by numbers, either 1, 2 or 3.
 	//
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:validation:Maximum=3
 	// +optional
-	AvailabilityZone string `json:"availabilityZone,omitempty"`
+	AvailabilityZone int32 `json:"availabilityZone,omitempty"`
```

Comment:
Can we change this to a string?

In CAPZ this is a string https://capz.sigs.k8s.io/reference/v1beta1-api#infrastructure.cluster.x-k8s.io/v1beta1.AzureMachineTemplateResource, as well as in Azure (https://learn.microsoft.com/en-us/rest/api/compute/resource-skus/list?view=rest-compute-2024-07-01&tabs=HTTP#resourceskulocationinfo) instead of an integer. I understand that it  currently maps to integers but if validation is desired that can be performed anyway with a string, but we would be consistent with what's in Azure we give space in case in the future changes.

Reply:
I had the same thought, I don't think we can have guarantees that azure will never introduce a different string that can't be mapped to a int. cc @JoelSpeed since you first introduce the change, any objections?

Reply:
The potential to not be an integer in the future is a fair one

What about using the kube IntOrString type? That would mean we don't have to force users to quote their integers, but, does it make validation harder? Perhaps

What validation can we impose now if we make this a string type, to ensure the zones provided are only `"1"`, `"2"` and `"3"` as we currently expect them to be?

Reply:
> What validation can we impose now if we make this a string type, to ensure the zones provided are only "1", "2" and "3" as we currently expect them to be?

Yes. In the future that validation then can then be relaxed if needed

Reply:
> What about using the kube IntOrString type? That would mean we don't have to force users to quote their integers, but, does it make validation harder? Perhaps

For me a string would be just fine. I don't see the need of coupling it to kube IntOrString or add more complexity. In Azure/CAPZ they don't do that.

Reply:
ok thanks! I choose to go with a string for simplicity, future flexibility while still keeping validation UX